PusherMainView/PusherMainView/FCMPusher.swift

```diff
@@ -0,0 +1,182 @@
+import Foundation
+
+protocol FCMPushable {
+    func pushUsingLegacyEndpoint(_ token: String,
+                                 payload: [String: Any],
+                                 collapseID: String?,
+                                 serverKey: String,
+                                 completion: @escaping (Result<String, Error>) -> Void)
+    func pushUsingV1Endpoint(_ token: String,
+                             payload: [String: Any],
+                             collapseID: String?,
+                             serverKey: String,
+                             projectID: String,
+                             completion: @escaping (Result<String, Error>) -> Void)
+}
+
+final class FCMPusher: FCMPushable {
+    let session = URLSession(configuration: .ephemeral,
+                             delegate: nil,
+                             delegateQueue: .main)
```

Comment:
nit: I think it can be interesting to instantiate `FCMPusher` with a session protocol so we can test FCMPusher with a session mock in `FCMPusherSpec`. I put this comment as nit because the PR is long enough to review. It can be done in an other PR :)